source/develop/release-management/features/network/allowExplicitDnsConfiguration.html.md

```diff
@@ -0,0 +1,132 @@
+authors: mmucha
+wiki_category: Feature
+wiki_title: Features/AllowExplicitDnsConfiguration
+wiki_revision_count: 12
+wiki_last_updated: 2015-10-14
+feature_name: Allow Explicit Dns Configuration
+feature_modules: Networking
+feature_status: Design
+rfe: https://bugzilla.redhat.com/show_bug.cgi?id=1160667
+
+---
+
+# Allow Explicit Dns Configuration
+
+### Owner
+
+*   Name: [ Martin Mucha](User:mmucha)
+*   Email: mmucha@redhat.com
+
+## Summary
+When new host is added to the system, management network will be created for it. Dns configuration for such network will be obtained from resolv.conf file. However admin should be able to specify overriding configuration in WebAdmin. He can do that at several places:
+
+* During creation new host
+* By updating management network of given host (in backend terms — via its NetworkAttachment)
+* By updating management network of given DC (or later of given Cluster) 
+
+## Detailed Description
+
+So DNS configuration can be stored in multiple places. If DNS configuration is not specified in either of them, no DNS configuration data will be passed to VDSM during SetupNetworks command and default values from resolv.conf will be used. If either of them is used, it will be passed to VDSM overriding configuration from resolv.conf. For example we can set DNS configuration in Network related to certain DC (or later to certain Cluster) and this configuration will override configuration in resolv.conf. If we now setup configuration in NetworkAttachment, attaching ManagementNetwork of given DataCenter (or later of given Cluster) to specific Host, this configuration is more specific than one in Network of DC (or later Cluster) scope and will be used instead.  
+
+### DB
+Database needs to be updated so it can accommodate dns configuration. Two places will be altered:
+
+* network_attachments table
+* network table
+
+both this tables must accommodate multiple DNS records. Naive solution would be to store them comma separated into single column. Better solution would be creating separate table for that. That's subject of further discussion.
+
+### REST
+
+As mentioned, you can specify DNS Configuration at three places. Corresponding REST areas will be altered. But first we need to add new element 'dns_configuration' as: 
+
+```
+<xs:element name="dns_configuration" type="DnsConfiguration"/>
+
+<xs:complexType name="DnsConfiguration">
+  <xs:complexContent>
+    <xs:extension base="BaseResource">
+      <xs:sequence>
+        <xs:element maxOccurs="1" minOccurs="0" name="ips" type="Ips"/>
+      </xs:sequence>
+    </xs:extension>
+  </xs:complexContent>
+</xs:complexType>
+```
+
+#### Creating new Host
+```
+POST /ovirt-engine/api/hosts HTTP/1.1
+Content-Type: application/xml
+
+<host>
+    <name>$NAME</name>
+    <address>$ADDRESS</address>
+    <port>$PORT</port>>
+    <root_password>$PASSWORD</root_password>
+    <cluster>
+        <name>$CLUSTERNAME</name>
+    </cluster>
+    <dns_configuration>
```

Comment:
> Will users ever read that? Most won't. 
really I wouldn't care about illiterate users. Any STOP sign on road also isn't surrounded by bubble foil  to save those, who can't read or are just to lazy to do so ;)

> Users may rely on that: an administrator may have manually and carefully configured the resolv.conffile, before adding the host, and these changes would be lost without warning if the management network has a custom DNS resolver configuration.
well _user_ of this rest is actually an admin. So if he edited the network, to contain dns configuration, he shouldn't be surprised, that this change takes place when adding new host. I don't assume, that there are lots of engine admins, so the one, who is adding new host is completely unaware of having management network changed. Therefore I have similar opinion as dan — I don't see need for having preserve_dns_resolver_configuration. Touching of dns configuration during adding new host will happen only for admin, and management network could have been also edited only by admin. And who else than admins should be familiar with system upgrade and its consequences...

> I believe that @mmucha-redhat meant that a management network attachment
yes, indeed. Sorry about confusion. Reading your discussion I feel, that because of all consequences and little usability, we can strip out manually setting dns configuration from adding a new host. We can live with having it set on network, from which it will be copied to network_attachment during adding new host. If admin needs to override some single host, he can in this exceptional case update management network - network attachment _[management network on one specific host]_ using separate call. If you agree, I'd remove usecase of setting dns configuration as a part of adding new host.